ZenPacks/zenoss/ZenPackLib/lib/spec/ZenPackSpec.py

```diff
@@ -466,6 +482,25 @@ def create_device_js_snippet(self):
         return self.create_js_snippet(
             'device', snippet, classes=device_classes)
 
+    def create_add_js_snippet(self):
+        """Register component add JavaScript snippet."""
+        device_classes = [
```

Comment:
Yes, that makes sense. Though it raises another point about only components that have a direct contained-by relationship to the device to be added in this way. It's worth thinking about what happens if someone enables "user_defined" on a component class that doesn't meet this criteria, or if they enable "user_defined" on a Device subclass.